src/wasm/literal.cpp

```diff
@@ -54,7 +57,9 @@ Literal& Literal::operator=(const Literal& other) {
       func = other.func;
       break;
     case Type::exnref:
-      new (&exn) auto(std::make_unique<ExceptionPackage>(*other.exn));
+      // Avoid calling the destructor, which may not be correct
+      new (&exn) auto(
+        std::move(std::make_unique<ExceptionPackage>(*other.exn)));
```

Comment:
This was already not calling the destructor; the addition of `std::move` is just a drive-by optimization.

It is important to _not_ call the destructor if the previous type was anything besides `exnref`, so this line was already correct. The only thing missing was that we had to call the destructor if the previous type _was_ `exnref`.

Reply:
This is one of those days where I think longingly of Rust 🦀